MetaMask/metamask-extension
# determine tx type using normalised tx params

## Explanation

Fixes issue where transaction data with omitted 0x will cause wrong transaction type (contract interaction when doing setApprovalForAll, this in turn is causing NFT transfer popup warning to not show up.

## Manual Testing Steps

- in test dapp run "set approval for all"
- you should see warning popup
```js
// should show warning confirmation screen after initial confirm
await window.ethereum.request({
  method: 'eth_sendTransaction',
  params: [{
    gas: '0xb403',
    value: '0x0',
    from: ethereum.selectedAddress,
    to: '0x39ec448b891c476e166b3c3242a90830db556661',
    data: '0xa22cb46500000000000000000000000039ec448b891c476e166b3c3242a90830db5566610000000000000000000000000000000000000000000000000000000000000001',
  }],
});

// should show warning confirmation screen after initial confirm, notice data is missisng 0x
await window.ethereum.request({
  method: 'eth_sendTransaction',
  params: [{
    gas: '0xb403',
    value: '0x0',
    from: ethereum.selectedAddress,
    to: '0x39ec448b891c476e166b3c3242a90830db556661',
    data: 'a22cb46500000000000000000000000039ec448b891c476e166b3c3242a90830db5566610000000000000000000000000000000000000000000000000000000000000001'
  }],
});
      
```
## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
